src/main/java/com/exasol/adapter/dialects/snowflake/SnowflakeSqlGenerationVisitor.java

```
@@ -0,0 +1,225 @@
+            break;
+        case WEEK:
+            builder.append("'WEEK'");
+            break;
+        case MONTH:
+            builder.append("'MONTH'");
+            break;
+        case YEAR:
+            builder.append("'YEAR'");
+            break;
+        default:
+            break;
+        }
+    }
+
+    private static void appendDecimalSize(ScalarFunction scalarFunction, StringBuilder builder) {
+        switch (scalarFunction) {
+        case SECOND:
+        case MINUTE:
+        case DAY:
+        case WEEK:
+        case MONTH:
+            builder.append("2");
+            break;
+        case YEAR:
+            builder.append("4");
+            break;
+        default:
+            break;
+        }
+    }
+
+    private String getPosixTime(final List<String> argumentsSql) {
+        return "EXTRACT(EPOCH FROM " + argumentsSql.get(0) + ")";
+    }
+
+    private String buildColumnProjectionString(final String typeName,DataType mappedType, final String projectionString) {
+        if (typeName.startsWith("NUMBER") && mappedType.getExaDataType() == DataType.ExaDataType.VARCHAR ){
+            return "'Number precision not supported'";
+        }
+        else if (typeName.startsWith("TIMESTAMPTZ")) {
+            return "TO_TIMESTAMP_NTZ(" + projectionString +")";
+        }
+        else if (checkIfNeedToCastToVarchar(typeName)) {
+            return "CAST(" + projectionString + "  as VARCHAR )";
+        } else if (typeName.startsWith("smallserial")) {
+            return "CAST(" + projectionString + "  as SMALLINT )";
+        } else if (typeName.startsWith("serial")) {
+            return "CAST(" + projectionString + "  as INTEGER )";
+        } else if (typeName.startsWith("bigserial")) {
+            return "CAST(" + projectionString + "  as BIGINT )";
+        } else if (TYPE_NAMES_NOT_SUPPORTED.contains(typeName)) {
+            return "cast('" + typeName + " NOT SUPPORTED' as varchar) as not_supported";
```

Would it be better to throw an exception here?

---

Removed this code. There's no serial types in snowflake.